lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
Using the host only is perhaps not quite enough to help with #7828, where it's needed to build a URL for say, Discovery - the protocol, port and path specified in :unattended_url are relevant too.

While a plethora of helpers is a bad thing, I can see a case for having helpers for either the FQDN and the base URL, _or_ the base URL only.  If the former, we can merge this as-is to provide the FQDN and re-open #7828 - I'm just unsure this fully solves it as-is.